### PR TITLE
v1.10 backports 2022-09-07

### PR DESCRIPTION
**Pull request by joamaki**
* #20112 -- Fix conflicting routes for multiple ENIs in IPAM mode (@recollir)
 * #20795 -- agent: Stop serving DNS before shutdown (@nebril)
 * #21133 -- operator: do not GC kvstore nodes if CiliumNodes are not available (@aanm)
 * #21146 -- datapath: tolerate missing ifaces when setting rp_filter sysctl (@julianwiedmann)
 * #21126 -- datapath: allow local NodePort traffic for `eni+` container interfaces with CNI chaining (@ti-mo)
 * #20848 -- Coalesce of health endpoint CIDRs (@dezmodue)
 * #21127 -- hubble-ui: release v0.9.2 (@geakstr)
 * #21202 -- operator: update CiliumNode in kvstore without lease (@tklauser)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 20112 20795 21133 21146 21126 20848 21127 21202; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label BRANCH=v1.10 ISSUES=20112,20795,21133,21146,21126,20848,21127,21202
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by joamaki** [comment]
Please check the following:
@nebril: #20795, No preCleanupFuncs in v1.10, had to change it to cleanupFuncs.Add().
@tklauser, @aanm: operator/cmd/cilium_node.go was operator/k8s_node.go in v1.10. Check if I ported changes correctly.
@ti-mo: Not sure that the CNIChainingMode commit did anything as it didn't seem to be used in v1.10.

File: daemon/cmd/ipam_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+//go:build !privileged_tests
```
Comment (tklauser) [inline daemon/cmd/ipam_test.go:4]: Looks like this needs an old `// +build` comment too given we still build `v1.10` using Go 1.17, see https://github.com/cilium/cilium/runs/8224150412?check_suite_focus=true#step:8:90

**Review by tklauser — APPROVED** [review]
Looks good for my change.

One comment re. Go build constraints inline.

File: daemon/cmd/ipam_test.go (hunk 1 above)
Comment (joamaki) [reply daemon/cmd/ipam_test.go:4]: Thanks, fixed.

**Review by julianwiedmann — APPROVED** [review]

**Review by geakstr — APPROVED** [review]

**Comment by tklauser** [comment]
> @tklauser leases `false` or `true` for `UpdateKeySync` from `clustermesh-apiserver/vmmanager.go`?
> 
> ```
> Error: level=warning msg="[runner] Can't run linter goanalysis_metalinter: err113: analysis skipped: errors in package: [/home/runner/work/cilium/cilium/clustermesh-apiserver/vmmanager.go:194:77: too few arguments in call to ciliumNodeRegisterStore.UpdateKeySync]"
> ```

The respective `UpdateSyncKey` call was removed in `master` by #18957 (commit 90c0d62c724469dee07853f8d104eaf167f9c24e) and replaced by a direct call to `kvstore.Client().UpdateIfDifferent` with `lease=true`. So I'd say for the matter of this backport `lease` should be `true` in the call to `UpdateKeySync` (or alternatively backport #18957 to `v1.10` as well).

File: pkg/option/config.go [hunk 2]
```diff
@@ -2569,6 +2575,7 @@ func (c *DaemonConfig) Populate() {
 	c.KeepConfig = viper.GetBool(KeepConfig)
 	c.KVStore = viper.GetString(KVStore)
 	c.KVstoreLeaseTTL = viper.GetDuration(KVstoreLeaseTTL)
+	c.CNIChainingMode = viper.GetString(CNIChainingMode)
```
Comment (ti-mo) [inline pkg/option/config.go]: Same as https://github.com/cilium/cilium/pull/21225#discussion_r965588268.

**Comment by ti-mo** [comment]
@joamaki Thanks for going through the trouble of fixing up the conflicts!

> @ti-mo: Not sure that the CNIChainingMode commit did anything as it didn't seem to be used in v1.10.

It's used in the follow-up commit. :wink:

**Review by nebril — APPROVED** [review]

**Comment by nebril** [comment]
/test-backport-1.10

**Comment by nebril** [comment]
/test-backport-1.10

**Review by ti-mo — APPROVED** [review]
Thanks!

**Pull request merged**